backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/llvm/IrToBitcode.kt

```diff
@@ -751,10 +771,24 @@ internal class CodeGeneratorVisitor(val context: Context, val lifetimes: Map<IrE
         }
 
         if (declaration.kind.isSingleton && !declaration.isUnit()) {
-            val value = context.llvmDeclarations.forSingleton(declaration).instanceFieldRef
-            LLVMSetInitializer(value, if (declaration.storageKind(context) == ObjectStorageKind.PERMANENT)
-                context.llvm.staticData.createConstKotlinObject(declaration,
-                        *computeFields(declaration)).llvm else codegen.kNullObjHeaderPtr)
+            val singleton = context.llvmDeclarations.forSingleton(declaration)
+            val access = singleton.instanceStorage
+            if (access is GlobalAddressAccess) {
+                LLVMSetInitializer(access.getAddress(null), if (declaration.storageKind(context) == ObjectStorageKind.PERMANENT)
+                    context.llvm.staticData.createConstKotlinObject(declaration,
+                            *computeFields(declaration)).llvm else codegen.kNullObjHeaderPtr)
+            }
+            // If can be exported and can be instantiated.
+            if (declaration.isExported() &&
+                    declaration.constructors.singleOrNull() { it.valueParameters.size == 0 } != null) {
```

Comment:
Isn't this true for any `object` declaration?

Reply:
No, that's why nightlies use to fall.

Reply:
Could you provide an example of `object` declaration with `declaration.constructors.singleOrNull() { it.valueParameters.size == 0 } == null`?

Reply:
PMed.